### DISCUSSION TITLE
【keyball44】USB 接続問題

### DISCUSSION THREAD

**Discussion by juncho26** [Yowkees/keyball#704] [Q&A]
最近勇者工房さんから Keyball44を購入し一か月間使っていたんですが、毎回パソコンをつけるたびにUSBケーブルを再接続しないとキーボールが使えないようです。多分ですがProMicroのLEDはついているので電源は入っているとは思いますが入力ができなくて、これは普通のことでしょうか？

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

KeyballにUSB給電されたタイミングでPCと通信できずにUSBデバイスと認識されないと、ご指摘のような状態となります。
PC環境にも依存しますが、ProMicroとQMKファームウェア使用の自作キーボード全般で起きる現象という認識でして、
申し訳ありませんが、仕様ということになってしまいます。

ただ私の知識も古いかも知れません。もしかすると解決策が出てきているかも知れませんので、ご存じの方いらっしゃいましたらご助言いただければ嬉しいです。

**Reply by NuuuDle13** [reply]
> ご連絡ありがとうございます。
> 
> KeyballにUSB給電されたタイミングでPCと通信できずにUSBデバイスと認識されないと、ご指摘のような状態となります。 PC環境にも依存しますが、ProMicroとQMKファームウェア使用の自作キーボード全般で起きる現象という認識でして、 申し訳ありませんが、仕様ということになってしまいます。
> 
> ただ私の知識も古いかも知れません。もしかすると解決策が出てきているかも知れませんので、ご存じの方いらっしゃいましたらご助言いただければ嬉しいです。


私のkeyball44でも同じ問題にぶつかりました。解決策を見つけました。私は、以下のコードを追加する必要がある。
 \keymaps\via\config.h

#define WAIT_FOR_USB

ファームウェアをコンパイルするが、出来上がったファイルは片方だけに書き込む。

両側に書き込むと、USBで接続されていない側は動作しません！

つまり、キーボードはメインの半分にのみUSB接続することができますが、問題は解決します。

どちらのハーフも別々に使用することができますが、両方のハーフを一緒に使用することは、UBSがメインハーフに接続されている場合にのみ機能します。

お幸せに :)

by deepl.com translator
